distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/ActionsBuilder.java

```diff
@@ -0,0 +1,53 @@
+package com.gruppo1.distributednetworkmanager;
+
+import com.dezen.riccardo.networkmanager.Resource;
+import com.dezen.riccardo.smshandler.Peer;
+
+/**
+ *
+ * @param <T> the real class that describes an Action of you custom network
+ * @param <P> a real type of peer that can be contacted
+ * @param <R> a resource type
+ * @param <N> a Node of the network
+ */
+public interface ActionsBuilder<T, P extends Peer, R extends Resource, N extends Node> {
+
+    T buildPing(P peer);
+
+
+    /**
+     * Hypothesis: this type of responses (ANSW) is built starting from the request (mainly to retrieve the action ID/number)
+     *
+     * QUESTION: is it good?
+     */
+    T buildPingAnsw(T request);
+
+
+    T buildInvite(P peer);
+
+    T buildInviteAnsw(T request, boolean accept);
+
+
+
+    T[] buildStore(P peer, P peerToStore);
```

Comment:
Why does this return an array? Case 1 of store works just like find node and value. If you find a good way to overload every method with the array counterpart I'd be fine with it (Maybe implicit arrays like P... would work, but I know for a fact they can be troublesome with generics, the issue should disappear once we implement the class with the actual types)